cookbook/agent_concepts/knowledge/filters/filtering_chroma_db.py

```diff
@@ -0,0 +1,85 @@
+from agno.agent import Agent
+from agno.knowledge.pdf import PDFKnowledgeBase
+from agno.utils.media import (
+    SampleDataFileExtension,
+    download_knowledge_filters_sample_data,
+)
+from agno.vectordb.chroma import ChromaDb
+
+# Download all sample CVs and get their paths
+downloaded_cv_paths = download_knowledge_filters_sample_data(
+    num_files=5, file_extension=SampleDataFileExtension.PDF
+)
+
+# Initialize ChromaDB
+vector_db = ChromaDb(collection="recipes", path="tmp/chromadb", persistent_client=True)
+
+# Step 1: Initialize knowledge base with documents and metadata
+# ------------------------------------------------------------------------------
+# When initializing the knowledge base, we can attach metadata that will be used for filtering
+# This metadata can include user IDs, document types, dates, or any other attributes
+
+knowledge_base = PDFKnowledgeBase(
+    path=[
+        {
+            "path": downloaded_cv_paths[0],
+            "metadata": {
+                "user_id": "jordan_mitchell",
+                "document_type": "cv",
+                "year": 2025,
+            },
+        },
+        {
+            "path": downloaded_cv_paths[1],
+            "metadata": {
+                "user_id": "taylor_brooks",
+                "document_type": "cv",
+                "year": 2025,
+            },
+        },
+        {
+            "path": downloaded_cv_paths[2],
+            "metadata": {
+                "user_id": "morgan_lee",
+                "document_type": "cv",
+                "year": 2025,
+            },
+        },
+        {
+            "path": downloaded_cv_paths[3],
+            "metadata": {
+                "user_id": "casey_jordan",
+                "document_type": "cv",
+                "year": 2025,
+            },
+        },
+        {
+            "path": downloaded_cv_paths[4],
+            "metadata": {
+                "user_id": "alex_rivera",
+                "document_type": "cv",
+                "year": 2025,
+            },
+        },
+    ],
+    vector_db=vector_db,
+)
+
+# Load all documents into the vector database
+knowledge_base.load(recreate=True)
+
+# Step 2: Query the knowledge base with different filter combinations
+# ------------------------------------------------------------------------------
+
+# Option 1: Filters on the Agent
```

Comment:
We can clean up if there is only 1 option

Reply:
yes, done, pushing